framework/tests/viewpointtest/VarargType.java

```diff
@@ -0,0 +1,9 @@
+public class VarargType {
+    void foo() {
+        TestClass a = new TestClass("testStr", new Object());
```

Comment:
> Also, add tests with 1) inner classes and 2) anonymous classes.
added
>Can you add test cases that actually have some viewpoint adaptation of the constructor, including tests with expected errors?
@wmdietl do you think it is better to open a new PR for viewpoint adaptation of constructor? Because this PR is trying to fix the crash error of using vararg type.

Reply:
This crash wasn't noticed because we didn't have the test case. It would be more convincing to have more test cases that actually show that constructors with varargs are now correctly adapted.

Reply:
Okay, I will add more tests as you commented below.